test/module/shared_model/backend_proto/proto_transaction_sequence_test.cpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gmock/gmock.h>
+#include <boost/range/adaptor/transformed.hpp>
+#include <boost/range/irange.hpp>
+#include "framework/result_fixture.hpp"
+#include "interfaces/iroha_internal/transaction_sequence.hpp"
+#include "module/shared_model/builders/protobuf/test_transaction_builder.hpp"
+
+using namespace shared_model;
+using ::testing::_;
+using ::testing::Return;
+using ::testing::Test;
+
+class MockTransactionCollectionValidator
+    : public validation::SignedTransactionsCollectionValidator<
+          validation::TransactionValidator<validation::FieldValidator,
+                                           validation::CommandValidatorVisitor<
+                                               validation::FieldValidator>>> {
+ public:
+  MOCK_CONST_METHOD1(
+      validatePointers,
+      validation::Answer(const interface::types::SharedTxsCollectionType &));
+};
+
+/**
+ * Creates transaction builder with set creator
+ * @return prepared transaction builder
+ */
+auto prepareTransactionBuilder(const std::string &creator) {
+  return TestTransactionBuilder().creatorAccountId(creator);
+}
+
+/**
+ * Creates atomic batch from provided creator accounts
+ * @param creators vector of creator account ids
+ * @return atomic batch of the same size as the size of creator account ids
+ */
+auto createAtomicBatch(std::vector<std::string> creators) {
```

Comment:
I assume, this function has a potential to be heavily used in future tests, so maybe move it to some kind of common files or something like that?

Reply:
And also, it would wise to parametrize it with an argument of type ```types::BatchType```, and create the corresponding batch type, so we wouldn't have two separate functions for ordered and atomic batches, with a lot of redundant code

Reply:
It should be put into the framework namespace. As you mentioned we can parameterise it with BatchType. But we might also need to set different creators. Or we might want to set transactions with certain commands. I would like to define possible use cases for such function and then rework it and put into separate file